demos/common/python/openvino/model_zoo/model_api/adapters/openvino_adapter.py

```diff
@@ -57,84 +57,103 @@ def __init__(self, ie, model_path, weights_path=None, device='CPU', plugin_confi
 
         self.model_from_buffer = isinstance(model_path, bytes) and isinstance(weights_path, bytes)
         log.info('Reading model {}'.format('from buffer' if self.model_from_buffer else model_path))
-        self.net = ie.read_network(model_path, weights_path, self.model_from_buffer)
+        weights = weights_path if self.model_from_buffer else ''
+        self.model = core.read_model(model_path, weights)
 
     def load_model(self):
-        self.exec_net = self.ie.load_network(self.net, self.device,
-            self.plugin_config, self.max_num_requests)
+        self.compiled_model = self.core.compile_model(self.model, self.device, self.plugin_config)
+        self.async_queue = AsyncInferQueue(self.compiled_model, self.max_num_requests)
         if self.max_num_requests == 0:
-            # ExecutableNetwork doesn't allow creation of additional InferRequests. Reload ExecutableNetwork
             # +1 to use it as a buffer of the pipeline
-            self.exec_net = self.ie.load_network(self.net, self.device,
-                self.plugin_config, len(self.exec_net.requests) + 1)
+            self.async_queue = AsyncInferQueue(self.compiled_model, len(self.async_queue) + 1)
 
         log.info('The model {} is loaded to {}'.format("from buffer" if self.model_from_buffer else self.model_path, self.device))
-        self.empty_requests = deque(self.exec_net.requests)
         self.log_runtime_settings()
 
     def log_runtime_settings(self):
         devices = set(parse_devices(self.device))
         if 'AUTO' not in devices:
             for device in devices:
                 try:
-                    nstreams = self.exec_net.get_config(device + '_THROUGHPUT_STREAMS')
+                    nstreams = self.compiled_model.get_config(device + '_THROUGHPUT_STREAMS')
                     log.info('\tDevice: {}'.format(device))
                     log.info('\t\tNumber of streams: {}'.format(nstreams))
                     if device == 'CPU':
-                        nthreads = self.exec_net.get_config('CPU_THREADS_NUM')
+                        nthreads = self.compiled_model.get_config('CPU_THREADS_NUM')
                         log.info('\t\tNumber of threads: {}'.format(nthreads if int(nthreads) else 'AUTO'))
                 except RuntimeError:
                     pass
-        log.info('\tNumber of network infer requests: {}'.format(len(self.exec_net.requests)))
+        log.info('\tNumber of model infer requests: {}'.format(len(self.async_queue)))
 
     def get_input_layers(self):
         inputs = {}
-        for name, layer in self.net.input_info.items():
-            inputs[name] = Metadata(layer.input_data.shape, layer.input_data.precision)
+        self.model_parameters['input_layouts'] = Layout.parse_layouts(self.model_parameters.get('input_layouts', None))
```

Comment:
You might just pass the `args.layout` to OpenvinoAdaper as `model_parameters` with default value `None`. The `parse_layouts()` anyway accepts the string, so the dictionary might be created after the string is parsed
```suggestion
        self.model_parameters = {'input_layouts': Layout.parse_layouts(self.model_parameters)}
```

Reply:
In my opinion, `model_parameters` should be dict. It may be useful for other parameters.